src/version/mod.rs

```diff
@@ -113,42 +107,56 @@ where
             option: self.option.clone(),
             timestamp: self.timestamp.clone(),
             log_length: self.log_length,
-            _p: Default::default(),
         }
     }
 }
 
-impl<R, FP> Version<R, FP>
+impl<R> Version<R>
 where
     R: Record,
-    FP: FileProvider,
 {
     pub(crate) async fn query(
         &self,
+        manager: &StoreManager,
         key: &TimestampedRef<R::Key>,
         projection_mask: ProjectionMask,
     ) -> Result<Option<RecordBatchEntry<R>>, VersionError<R>> {
+        let level_0_path = self
+            .option
+            .level_fs_path(0)
+            .unwrap_or(&self.option.base_path);
+        let level_0_fs = manager.get_fs(level_0_path);
         for scope in self.level_slice[0].iter().rev() {
             if !scope.contains(key.value()) {
                 continue;
             }
             if let Some(entry) = self
-                .table_query(key, &scope.gen, projection_mask.clone())
+                .table_query(level_0_fs, key, &scope.gen, projection_mask.clone())
                 .await?
             {
                 return Ok(Some(entry));
             }
         }
-        for level in self.level_slice[1..6].iter() {
-            if level.is_empty() {
+        for (i, sort_runs) in self.level_slice[1..6].iter().enumerate() {
```

Comment:
should it be `level_slice[1..MAX_LEVEL]`? level_slice[6] is not included in this case